### ISSUE TITLE
Variable not enclosed in {} AND single quoted string containing a variable

### ISSUE THREAD

**Issue by krissik** [rodjek/puppet-lint#434]
If I do two things wrong like this:

```
$var = 'foo'
exec {'bar $var':
  command => 'ls -alh'
}
```

puppet-lint doesn't warn me about "variable not enclosed in {} on line 2" and "single quoted string containing a variable".

I am using puppet-lint 1.1.0 and puppet 3.4.3.

**Comment by solarkennedy** [comment]
There are legitimate use cases where you need a literal `$var`, for which using a single quoted string is what you really mean. Take for example using the getvar function:

```
getvar('$::somefact')
```

I understand that there is a blurry line between linting and catching bugs. Yes if such a check existing I guess I would simply disable it when I needed to use getvar or defined or needed a literal `$` in a bash script or nginx proxy directive, etc.

**Comment by rnelson0** [comment]
As described, this is operating as expected. If the contents instead say`exec {'bar ${var}':`, it then triggers the check `ERROR: single quoted string containing a variable found on line 2`.
